[FX] Add in resnet + quantization tests

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43157 [FX] Add in resnet + quantization tests**

Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

Commit:
[FX] Add in resnet + quantization tests

[ghstack-poisoned]

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]

zdevito commented on torch/fx/quantization.py:
```diff
@@ -0,0 +1,329 @@
+r'''
```
We should put this in the test folder similar to `test_jit.py` that has its own subfile. Otherwise someone might see it and think it has to do with real quanitization.

zdevito commented on torch/fx/quantization.py:
```diff
@@ -0,0 +1,329 @@
+        self.quantized_graph = Graph()
+        self.delegate = DelegateBase(self.quantized_graph)
+
+        env = {}
```
This should make `name -> Node` instead of `name -> Proxy`. That will remove the need for `_create_proxy` in almost all cases where it is used.

zdevito commented on torch/fx/quantization.py:
```diff
@@ -0,0 +1,329 @@
+        def load_arg(n, quantized):
+            if not quantized:
+                if n.name not in env and n.name in quant_env:
+                    env[n.name] = quant_env[n.name].dequantize()
```
This will be the only place where you need to construct a proxy `Proxy(quant_env[n.name]).dequantize().node`

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]

Commit:
Update on "[FX] Add in resnet + quantization tests"


Differential Revision: [D23173327](https://our.internmc.facebook.com/intern/diff/D23173327)

[ghstack-poisoned]
